models/models.go

```diff
@@ -0,0 +1,72 @@
+package models
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+)
+
```

Comment:
There are a few features that comprise a good design. Morevoer, they can be made more speficic given the relational nature of the utilized database. In general these features are : scalability, maintainability, reliability, security and ofcourse performance. 

I'll describe what these features encompass in the context of a relational database. 
- Scalability is supported by creating efficient database schemas, using appropriate indices and where/when needed using advanced scaling techniques such as sharding. 
- Maintainability is the ease with which the application can be updated to include new features or if requirements change. This relies heavily on good code organization through the use of clear design patterns.
- Reliability is focused around making sure that the application does not have unexpected behaviour (crash, freeze, unhandled events, etc...) as well as preventing any data loss. This feature is best addressed on both the application and the platform level.
- Security should be one of the main focus points of the application, especially when dealing with user data. Even more so if the data are extremely sensitive (financial, medical, etc...). Nowadays, the bare minimum should be authorization,authentication and encryption mechanisms.
- Performance of the application is one of the most coveted targets and its importance is exacerbated when dealing with massive amount of users. Typical optimization techniques that would lead to low response times in this case are optimized queries to the database and use of caching to reduce database operations.

The typical steps to accomplishing all these features, and in the desired degree, include: 
- a clear definition of the application's requirements
- definition of the database schema (minimizing as much as possible redundancy and utilizing as much as possible relationships) and creation of the database taking into consideration security aspects.
- Develop the application making use of appropriate design patterns.
- Optimize the db queries that the application makes.
- Examine the application's behaviour in order to detect optimization opportunities. This is typically a combination of profiling and monitoring.
- Make sure that security consideration are honored in all cases. A part of the security/reliability considerations should always be a backup/recovery strategy.

Reply:
I guess that would be normalization. Now, which form to follow though would be dependent on the requirements, while keeping in mind that higher forms lead to higher db complexity thus impacting the maintainability i mentioned earlier (and possibly the performance).